### PR TITLE
use context for search state and don't persist it in the URL

### PR DESCRIPTION
**Pull request by jamesgorrie**
- Creates a global context shared from `_app` between all pages.
- The context get's its initial state from the router, and updates `onRouteChange`.
- The context is the updated from `works`, and can be read on the `work` or `item` page. An example if the back to search and search box on the `work` page.
- Context is forgotten when you refresh the app... where this could be a problematic things from a UX POV is if I search for `hats`, go to a result, and then refresh the page, the search bar will forget my preference. This can be mitigated later by having a separate store e.g. sessionStorage is it surfaces as a problem.
- Removed quite a lot of logic around defaults that is no longer relevant and makes the code quite hard to fathom. We will need to remove the rest of the `booksRelease` toggle too.

### CODE REVIEW

**Comment by jamesgorrie** [comment]
@wellcometrust/experience-devs all the kinks are ironed out and ready for review.

File: common/views/components/CatalogueSearchContext/CatalogueSearchContext.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    queryType,
+    setQuery,
+    setPage,
+    setWorkType,
+    setItemsLocationsLocationType,
+    setQueryType,
+  };
+
+  useEffect(() => {
+    function routeChangeComplete(url: string) {
```
Comment (davidpmccormick) [inline common/views/components/CatalogueSearchContext/CatalogueSearchContext.js]: Does this function have to live inside `useEffect`?

**Comment by davidpmccormick** [comment]
Does/could this help with #4342 ?

File: common/views/components/CatalogueSearchContext/CatalogueSearchContext.js (hunk 1 above)
Comment (jamesgorrie) [reply common/views/components/CatalogueSearchContext/CatalogueSearchContext.js]: This should only run on the client, so yip

**Comment by jamesgorrie** [comment]
> Does/could this help with #4342 ?

@davidpmccormick I would have thought so, but doesn't, we can look at why in another PR.

**Pull request merged**